presto-main/src/main/java/com/facebook/presto/EventListenerManager.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto;
+
+import com.facebook.presto.spi.eventlistener.EventListener;
+import com.facebook.presto.spi.eventlistener.EventListenerFactory;
+import com.facebook.presto.spi.eventlistener.QueryCompletedEvent;
+import com.facebook.presto.spi.eventlistener.QueryCreatedEvent;
+import com.facebook.presto.spi.eventlistener.SplitCompletedEvent;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.Maps.fromProperties;
+import static java.util.Objects.requireNonNull;
+
+public class EventListenerManager
+{
+    private static final File EVENT_LISTENER_CONFIGURATION = new File("etc/event-listener.properties");
+
+    private final ConcurrentMap<String, EventListener> eventListeners = new ConcurrentHashMap<>();
+
+    public void addEventListenerFactory(EventListenerFactory eventListenerFactory)
+    {
+        requireNonNull(eventListenerFactory, "eventListenerFactory is null");
+
+        Map<String, String> properties = ImmutableMap.of();
+        try {
+            if (EVENT_LISTENER_CONFIGURATION.exists()) {
+                properties = ImmutableMap.copyOf(loadProperties(EVENT_LISTENER_CONFIGURATION));
+            }
+        }
+        catch (Exception ignored) {
```

Comment:
Also, having a single properties file for all event listener plugins is going to cause problems due to unused property validations. Either we allow a single listener to be configured, or we make it possible to provide per-listener configuration files.